feat(command): adding args to use line

danmx opened the pull request:
Adding `args` to use line.


Old:
```
❯ ./test
Error: accepts 1 arg(s), received 0
Usage:
  test [flags]

Flags:
  -h, --help     help for test
  -t, --toggle   Help message for toggle
  ```
  
  New:
  ```
  ❯ ./test
Error: accepts 1 arg(s), received 0
Usage:
  test [flags] [args]

Flags:
  -h, --help     help for test
  -t, --toggle   Help message for toggle
  ```
  
  Diff:
  ```diff
3c3
<   test [flags]
---
>   test [flags] [args]
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spf13/cobra?pullRequest=1210) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/spf13/cobra?pullRequest=1210) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spf13/cobra?pullRequest=1210) it.</sub>

danmx commented:
@jpmcb could you take a look?

umarcor commented:
Ref #842

jpmcb commented:
Hi @danmx thanks for this PR - would you be able to add a test for this? Maybe in `command_test.go`? A passing test will help this get reviewed and merged in

danmx commented:
@jpmcb took me some time but I've added tests for `UseLine`.

danmx commented:
@marckhouzam I see that you were active recently. Could you take a look at this PR?

I addressed the previous comment about adding tests.
I had to reshuffle `Command` struct optimal memory usage (fixing violations of `maligned` linter) because I added another bool field.

marckhouzam commented:
@danmx could you show the change in output that this provides in the description of the PR so I can know what to expect?

danmx commented:
@marckhouzam the change is:

Old:
```
❯ ./test
Error: accepts 1 arg(s), received 0
Usage:
  test [flags]

Flags:
  -h, --help     help for test
  -t, --toggle   Help message for toggle
  ```
  
  New:
  ```
  ❯ ./test
Error: accepts 1 arg(s), received 0
Usage:
  test [flags] [args]

Flags:
  -h, --help     help for test
  -t, --toggle   Help message for toggle
  ```
  
  Diff:
  ```diff
3c3
<   test [flags]
---
>   test [flags] [args]
```

marckhouzam commented:
Could you not reorder the fields of the command structure? Although your order may be better it makes the review harder. It’s hard for me to find time for reviews so if you keep the PR as small as possible it’ll increase your chances of getting it merged.

marckhouzam commented:
> test [flags] [args]

I don’t think we can do this. The `command.Use` field is defined by the program and many programs specify the arguments the command expects in their Use field. In such cases the new `[args]` would be repeating.

For example https://github.com/helm/helm/blob/a3c903e4c62839d24389a1e0cdf550a8d759f054/cmd/helm/pull.go#L49
